### PR TITLE
Drivers

### PR DESCRIPTION
**Pull request by gaetan-cordonnier**
Component Drivers with modal is functional.

### CODE REVIEW

**Review by Guy-deMessimy — DISMISSED** [review]

**Review by gaetan-cordonnier — COMMENTED** [review]
Conflicts resolved.

**Review by Guy-deMessimy — DISMISSED** [review]

**Review by AmjadLille — DISMISSED** [review]

**Review by Guy-deMessimy — DISMISSED** [review]

File: public/images/animated_background.svg [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            @keyframes float1 {
```
Comment (LoicBrassart) [inline public/images/animated_background.svg]: Wait, what ? No !
CSS goes in a separate file !

File: src/components/Drivers/Modal/Data.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const driverItems = [
```
Comment (LoicBrassart) [inline src/components/Drivers/Modal/Data.jsx:1]: This object is futilely complex, you could simplify it a lot!
- Don't open a global object in the array
- Don't add artificial IDs as keys
=> You have an array, with every element being an Object, even IDs can be kept in them if needed

File: src/components/Drivers/Modal/index.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import driverItems from "./Data";
+import { ModalStyle } from "./style";
+
+export default function Modal({ itemNumber }) {
+  console.log(itemNumber);
```
Comment (LoicBrassart) [inline src/components/Drivers/Modal/index.jsx]: No `console.log`

File: src/components/Drivers/index.jsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+import { DriversStyle } from "./style";
+import Modal from "./Modal/index";
+
+export default function Drivers() {
+  const [isOpen, setIsOpen] = useState(false);
+  let [driverItem, setDriverItem] = useState();
+
+  const hexagonSize = { x: 10, y: 10 };
+
+  console.log(isOpen);
```
Comment (LoicBrassart) [inline src/components/Drivers/index.jsx]: Neither here

File: src/components/Drivers/index.jsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import { HexGrid, Layout, Hexagon, Text, Pattern } from "react-hexgrid";
+import { DriversStyle } from "./style";
+import Modal from "./Modal/index";
+
+export default function Drivers() {
+  const [isOpen, setIsOpen] = useState(false);
+  let [driverItem, setDriverItem] = useState();
```
Comment (LoicBrassart) [inline src/components/Drivers/index.jsx]: Why use `let` ?

File: src/components/Drivers/index.jsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          origin={{ x: -0, y: -20 }}
+        >
+          <Hexagon className="hexBlue" q={0} r={0} s={0} fill="pat-1"></Hexagon>
+          <Hexagon
+            className="hexYellow"
+            q={-1}
+            r={0}
+            s={0}
+            onMouseEnter={() =>
+              setIsOpen(true, setDriverItem((driverItem = 1)))
```
Comment (LoicBrassart) [inline src/components/Drivers/index.jsx]: `driverItem = 1` ? That probably does not do what you think it does

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Please refactor this code

File: src/components/Drivers/index.jsx (hunk 6 above)
Comment (gaetan-cordonnier) [reply src/components/Drivers/index.jsx]: "driverItem = 1" -> It's correct this is the number of Hexagon Index

File: public/images/animated_background.svg (hunk 1 above)
Comment (gaetan-cordonnier) [reply public/images/animated_background.svg]: Resolved

File: src/components/Drivers/Modal/index.jsx (hunk 3 above)
Comment (gaetan-cordonnier) [reply src/components/Drivers/Modal/index.jsx]: Resolved

File: src/components/Drivers/index.jsx (hunk 4 above)
Comment (gaetan-cordonnier) [reply src/components/Drivers/index.jsx]: It's correct, it's for the library

File: src/components/Drivers/index.jsx (hunk 5 above)
Comment (gaetan-cordonnier) [reply src/components/Drivers/index.jsx]: Resolved

File: src/components/Drivers/index.jsx (hunk 6 above)
Comment (gaetan-cordonnier) [reply src/components/Drivers/index.jsx]: Resolved

File: public/images/animated_background.svg (hunk 1 above)
Comment (gaetan-cordonnier) [reply public/images/animated_background.svg]: Image deleted on folder

**Review by AmjadLille — DISMISSED** [review]

**Review by AmjadLille — DISMISSED** [review]

File: src/components/Drivers/index.jsx (hunk 4 above)
Comment (LoicBrassart) [reply src/components/Drivers/index.jsx]: Nope, no `console.log` is correct

**Review by AmjadLille — DISMISSED** [review]

**Review by AmjadLille — DISMISSED** [review]

**Review by Guy-deMessimy — DISMISSED** [review]

**Review by LoicBrassart — DISMISSED** [review]
SnE

**Review by Guy-deMessimy — DISMISSED** [review]

**Review by AmjadLille — APPROVED** [review]

**Review by Guy-deMessimy — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
SnE

**Pull request merged**